### DISCUSSION TITLE
Need help with menubar primitive

### DISCUSSION THREAD

**Discussion by adrian-novacescu** [radix-ui/primitives#2529] [Help]
Hi, 

I used this component in order to create a tree selector. Everything except scrolling works ok. I used the same approach I used for Select primitive when I wrapped it in a scroll view but without succes. The code I did is the one bellow: 

` <Menubar.Root className={className}>
      <Menubar.Menu>
        <TreeSelectorTrigger filterable placeholder={placeholder} />
        <Menubar.Portal>
          <ScrollArea.Root type="auto" className="tw-w-full tw-h-full">
            <Menubar.Content className="tw-bg-white tw-rounded-md tw-drop-shadow-lg tw-shadow-inner focus-visible:tw-outline-none focus-visible:tw-ring-0">
              <ScrollArea.Viewport
                className="tw-w-full tw-h-[300px] tw-max-h-radixSelectContent"
                style={{ overflowY: undefined }}
              >
                <TreeSelectorContent
                  onSelect={({ name, checked }) => {
                    onSelect({ treeKey: name, selected: checked });
                  }}
                  selectorContentId={id}
                  tree={selectorTree}
                />
              </ScrollArea.Viewport>
              <ScrollArea.Scrollbar
                className="tw-w-2.5 tw-px-0.5"
                orientation="vertical"
              >
                <ScrollArea.Thumb className="tw-bg-gray-400 tw-rounded tw-cursor-pointer" />
              </ScrollArea.Scrollbar>
            </Menubar.Content>
          </ScrollArea.Root>
        </Menubar.Portal>
      </Menubar.Menu>
    </Menubar.Root>`

The result I am looking for is:
<img width="813" alt="Screenshot 2023-11-16 at 20 15 23" src="https://github.com/radix-ui/primitives/assets/36935575/6e5eec1a-aedb-4445-8cba-241eb85f599a">

However, in order to make it work I set scrollView height to 300px. On select primitive I used height: 100% and the select component took whole available space. Discussions about that implementation can be seen on https://github.com/radix-ui/primitives/discussions/2272#discussioncomment-6452451

I am looking forward to your responses/suggestions.

Thanks a lot,
Adi

**Comment by joaom00** [comment]
Sorry, I couldn't understand what the problem was with the scroll. Can you elaborate further? A reproducible example would be of great help as well.

**Comment by adrian-novacescu** [comment]
Thank you for your answer. I can create an example next week. The problem is that ScrollArea height is identical with its content height. It doesn't respect the max-height = tw-max-h-radixSelectContent. I fixed it by adding a ref and computing the max size myself but maybe there is a better way?

In addition, I just found a new issue now. I use an input as a trigger. I need it in order to filter out options from the tree. However, space is not working because the trigger intercept the event and opens the menu. Do you have any suggestions here?

Next week I will come back with a sandbox where I will try to reproduce both cases. 

Thanks again, 
Adi

**Reply by joaom00** [reply]
Just to be sure, What's in the `tw-max-h-radixSelectContent` class?

**Comment by adrian-novacescu** [comment]
I have maxHeight: {
        radixSelectContent:
          'var(--radix-select-content-available-height, 100%)',
      } in tailwind config. However, --radix-select-content-available-height seems to be an empty string all the time.

**Reply by joaom00** [reply]
Well, `--radix-select-content-available-height` is only for `Select` primitive and you are using `Menubar`

**Comment by adrian-novacescu** [comment]
Correct. Sorry about that :( and thanks a lot :).
However, if I am thinking trough, I can't use `--radix-menubar-content-available-height` either since I have multiple submenus open at the same time so this variable will be conflicting between them right?

Any suggestion for the other issue I have? The one where I try to use an input as trigger and make it WCAG compatible?

**Reply by joaom00** [reply]
> However, if I am thinking trough, I can't use --radix-menubar-content-available-height either since I have multiple submenus open at the same time so this variable will be conflicting between them right?

Actually, I'm not sure... :sweat_smile: 

> Any suggestion for the other issue I have? The one where I try to use an input as trigger and make it WCAG compatible?

What are you trying to build? something like a combobox? Any reason to use menubar instead of dropdown menu? I would say that no primitive in Radix works well with input as a trigger without workarounds... your best bet is to try to use another library for this.

**Comment by adrian-novacescu** [comment]
Sorry it took so long to respond. I have a sandbox on https://codesandbox.io/s/treeselector-95fp8f. However, it works a bit worse than it works on our product where I can select options using the keyboard.
Basically we need that in order to set a thema code (international standard for a book topic) to a book. We need to be able to select anything from a broad category until a specific topic so we came up with a treeSelector. I tried to use the dropdown menu instead of menubar but when I was hovering on a submenu, the whole menu closed and I was unable to identify why.

Looking forward tu your response,
Adi
